helix-term/src/ui/spinner.rs

```diff
@@ -32,7 +42,7 @@ pub struct Spinner {
 impl Spinner {
     /// Creates a new spinner with `frames` and `interval`.
     /// Expects the frames count and interval to be greater than 0.
-    pub fn new(frames: Vec<&'static str>, interval: u64) -> Self {
+    pub fn new(frames: Vec<String>, interval: u64) -> Self {
         let count = frames.len();
         assert!(count > 0);
         assert!(interval > 0);
```

Comment:
Now that this is configurable, we should remove these assertions and not render the spinner if either `count` or `interval` is 0.

Reply:
Great idea. Done.